### ISSUE TITLE
Truth Social giving Error 403

### ISSUE THREAD

**Issue by jason-oliveira** [yt-dlp/yt-dlp#7001]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that a **supported** site is broken
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Provide a description that is worded well enough to be understood

Truth Social is providing Error 403 to people attempting to pull rumble-sourced pages from Truth Social feeds using yt-dlp for the last two weeks. 

However, the testing on this might be difficult. Without getting into too many details, I have made a private agreement with Truth Social and its parent organizations to allow my VPN's US IP addresses to be whitelisted, to facilitate access without creating a user account. As I am not residing in the United States, I am unaware of and unable to test default behavior for unregistered access to Truth Social posts in the United States. Triaging users with legitimate US IP address receiving 403 errors vs my setup might be important.

However, should you not be able to replicate unregistered access 403 Forbidden errors, meaning terms of a private agreement have been modified unilaterally for my whitelisted IP addresses, this would be important data to have as well.

Another caveat to testing: I have seen high-volume accounts where approximately 30-80% of the posts during a day will be accessible via yt-dlp. it may require monitoring a high-volume account such as listed in the verbose output.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://truthsocial.com/@KarliBonne/posts/110319486893480616']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.10.10 (CPython x86_64 64bit) - Linux-6.2.5-gentoo-dist-x86_64-AMD_Ryzen_3_3200U_with_Radeon_Vega_Mobile_Gfx-with-glibc2.36 (OpenSSL 1.1.1t  7 Feb 2023, glibc 2.36)
[debug] exe versions: ffmpeg 4.4.3 (fdk,setts), ffprobe 4.4.3, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-3021.03.16, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[Truth] Extracting URL: https://truthsocial.com/@KarliBonne/posts/110319486893480616
[Truth] 110319486893480616: Downloading JSON metadata
ERROR: [Truth] 110319486893480616: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 694, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/truth.py", line 55, in _real_extract
    status = self._download_json(f'https://truthsocial.com/api/v1/statuses/{video_id}', video_id)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1048, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1012, in download_handle
    res = self._download_webpage_handle(
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 882, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 839, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 821, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3742, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

**Comment by bashonly** [comment]
You're not experiencing the 403 issue because you have a private agreement with the website, but you want us to test it and report back data to you? This is not what this issue tracker is for. If you have a direct line to the company, can't you sort this out?

**Comment by jason-oliveira** [comment]
no, that's not it at all, and it's depressing that you closed this bug prematurely.

Have you even attempted to replicate the URL listed?

As you can see in the results, there is an Error 403 happening. 
A user is reporting an Error 403 Forbidden.
If this is not occurring with other users, that's fine.
But closing a bug without triage? and marking it invalid?

I get you don't like useless bug reports, but this is a valid issue, and the "hose-and-close" on the bug report, with implicit threat of banning and/or punishment if re-reporting, is pretty disgusting.

**Comment by bashonly** [comment]
```
$ yt-dlp -F 'https://truthsocial.com/@KarliBonne/posts/110319486893480616'
[Truth] Extracting URL: https://truthsocial.com/@KarliBonne/posts/110319486893480616
[Truth] 110319486893480616: Downloading JSON metadata
[info] Available formats for 110319486893480616:
ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
0  qt  unknown    │ https │ unknown unknown
```

**Comment by jason-oliveira** [comment]
Excellent. Thank you for confirming it's local-only with a WORKSFORME.
Have a fantastic day.

**Comment by gamer191** [comment]
> I get you don't like useless bug reports, but this is a valid issue

It was closed because it appears that you have a private deal with Truth Social, and that that private deal is the reason for your issue (which no one can reproduce). If that is the case, it is unclear why you would request support from us

> with implicit threat of banning and/or punishment if re-reporting

We'd probably be annoyed if you opened a new issue (as opposed to simply posting a comment), but, whilst that decision isn't up to me, I'm certain you wouldn't be banned

**Comment by pukkandan** [comment]
> * [x]  I've checked that all provided URLs are playable in a browser with the same IP and same login details

From your post it doesn't sound like you have actually done this, have you? It's not our responsibility to triage your agreements with a third party

**Comment by jason-oliveira** [comment]
@pukkandan confirmed. URL is playable in-browser (Brave v1.51.110). I have no account on Truth Social, so no login is used. I have no access to Truth Social profiles or posts not marked public. To be clear, the responsibility is to determine whether a user is experiencing a reproducible error that would affect other users. That is what is currently unclear at this point, as I am the only one experiencing it, presumably, and @bashonly having proved that it worked without issue on their side.

Interestingly, it works with `--print-traffic -v`, but then stops working again when flags are not used. 

Log request attached:

[Truth Social giving Error 403 · Issue #7001 log request.txt](https://github.com/yt-dlp/yt-dlp/files/11413484/Truth.Social.giving.Error.403.Issue.7001.log.request.txt)

@gamer191 @bashonly's confirmation of functionality was sufficient for me to begin communication with Truth Social, which is why I thanked him once he proved it to work after closing. I'd rather report a bug and risk developer ridicule for wasting time than leave a project I use and support with broken functionality.

**Comment by gamer191** [comment]
> Interestingly, it works with --print-traffic -v, but then stops working again when flags are not used.

That makes no sense. Can you please run it with `--print-traffic -v` until it fails and then send the log.

On a side note, I'm curious if you can access this url (in a browser): https://truthsocial.com/api/v1/statuses/110319486893480616

**Comment by gamer191** [comment]
After further debugging I can reproduce this with github actions, suggesting that only some IP addresses are affected

Anyway, the issue is that cloudflare are detecting yt-dlp (similar to https://github.com/yt-dlp/yt-dlp/issues/6748). It will be a while until that can be solved, but in the meantime you can avoid their detection using `--legacy-server-connect`

**Comment by someziggyman** [comment]
Just will paste a log with the same issue below in case more info is needed for testing:

./yt-dlp -Uv -F "https://truthsocial.com/@NaturalHealing/posts/113626403717360921"
[debug] Command-line config: ['-Uv', '-F', 'https://truthsocial.com/@NaturalHealing/posts/113626403717360921']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.12.06 from yt-dlp/yt-dlp [4bd265539] (zip)
[debug] Python 3.12.2 (CPython arm64 64bit) - macOS-14.7.1-arm64-arm-64bit (OpenSSL 3.0.13 30 Jan 2024)
[debug] exe versions: phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.1, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, curl_cffi
[debug] Extractor Plugins: SamplePluginIE
[debug] Post-Processor Plugins: SamplePluginPP
[debug] Loaded 1961 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.12.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.12.06 from yt-dlp/yt-dlp)
[Truth] Extracting URL: https://truthsocial.com/@NaturalHealing/posts/113626403717360921
[Truth] 113626403717360921: Downloading JSON metadata
ERROR: [Truth] 113626403717360921: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: Forbidden>)
  File "./yt-dlp/yt_dlp/extractor/common.py", line 742, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/truth.py", line 54, in _real_extract
    status = self._download_json(f'https://truthsocial.com/api/v1/statuses/{video_id}', video_id)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/common.py", line 1152, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/common.py", line 1112, in download_handle
    res = self._download_webpage_handle(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/common.py", line 962, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/common.py", line 911, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "./yt-dlp/yt_dlp/extractor/common.py", line 898, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query, extensions))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/YoutubeDL.py", line 4166, in urlopen
    return self._request_director.send(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/networking/common.py", line 117, in send
    response = handler.send(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/networking/_helper.py", line 208, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/networking/common.py", line 340, in send
    return self._send(request)
           ^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/networking/_requests.py", line 365, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden

**Comment by gamer191** [comment]
I'm getting this error now. The solution is `--impersonate chrome`

**Comment by TheFiZi** [comment]
I don't know if this is helpful but if I use yt-dlp on my Mac, I can download videos from Truth Social. If I try to do it via a Linux VM it fails.

Working, MacBook Air M4:

```
[debug] Command-line config: ['https://truthsocial.com/@realDonaldTrump/posts/115670539534836133', '-v', '--print-traffic']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.11.12 from yt-dlp/yt-dlp [335653be8] (pip)
[debug] Python 3.13.10 (CPython arm64 64bit) - macOS-26.1-arm64-arm-64bit-Mach-O (OpenSSL 3.6.0 1 Oct 2025)
[debug] exe versions: ffmpeg 8.0.1 (setts), ffprobe 8.0.1
[debug] Optional libraries: sqlite3-3.51.1
[debug] JS runtimes: none
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Plugin directories: none
[debug] Loaded 1844 extractors
[Truth] Extracting URL: https://truthsocial.com/@realDonaldTrump/posts/115670539534836133
[Truth] 115670539534836133: Downloading JSON metadata
director: Handler preferences for this request: urllib=0
director: Checking if "urllib" supports this request.
director: Sending request via "urllib"
send: b'GET /api/v1/statuses/115670539534836133 HTTP/1.1\r\nHost: truthsocial.com\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/137.0.0.0 Safari/537.36\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Language: en-us,en;q=0.5\r\nSec-Fetch-Mode: navigate\r\nAccept-Encoding: gzip, deflate\r\nConnection: close\r\n\r\n'
reply: 'HTTP/1.1 200 OK\r\n'
header: Date: Sat, 06 Dec 2025 06:03:39 GMT
header: Content-Type: application/json
header: Content-Length: 1126
header: Connection: close
header: CF-RAY: 9a9994d3bb41051c-YVR
header: Content-Encoding: gzip
header: vary: Accept-Encoding
header: content-security-policy: base-uri 'none'; default-src 'none'; font-src 'self'; img-src 'self' https: data: blob: https://static-assets-1.truthsocial.com; media-src 'self' https: data: blob: https://static-assets-1.truthsocial.com; style-src 'self' 'unsafe-inline'; frame-src 'self' https:; manifest-src 'self'; connect-src 'self' data: blob: wss://truthsocial.com https://truthsocial.com wss://links.truthsocial.com https://links.truthsocial.com wss://uploads.truthsocial.com https://uploads.truthsocial.com https://*.rmbl.ws https://*.ingest.sentry.io https://*.luckyorange.com https://pubsub.googleapis.com wss://*.visitors.live wss://*.luckyorange.com *.mixpanel.com *.mxpnl.com https://*.revcontent.com https://*.criteo.com https://*.criteo.net https://challenges.cloudflare.com https://*.perception.tv https://vstream.truthsocial.com https://*.truthsocial.tv https://*.googlesyndication.com https://*.adtrafficquality.google https://payments.braintree-api.com/graphql https://api.braintreegateway.com/* https://consent.cookie-script.com; script-src 'self' https://tools.luckyorange.com https://*.mxpnl.com https://assets.revcontent.com https://polyfill.io https://ads.pubmatic.com https://*.criteo.com https://challenges.cloudflare.com https://*.googlesyndication.com https://connect.facebook.net https://*.us.tvsquared.com https://googletagmanager.com https://*.adtrafficquality.google https://cdn.cookie-script.com; child-src 'self'; worker-src 'self' blob:; frame-ancestors 'none'
header: cf-cache-status: DYNAMIC
header: Set-Cookie: __cf_bm=bfF7E1sNcPdX0SWIVtJgsJQX.PW7EcagJyt2dTfwC_s-1765001019-1.0.1.1-1bEo4U0EjUB7_dcjCH2V25pbLX9JblYdq0y7KwtKRoMHXmRAIOySGT_dWpjg_nUMdvJkNd.z8hhG0Q9wMRnm08_zeNhWhC5x6WbphxVeRyI; path=/; expires=Sat, 06-Dec-25 06:33:39 GMT; domain=.truthsocial.com; HttpOnly; Secure; SameSite=None
header: Set-Cookie: __cflb=0H28vTPqhjwKvpvovPffsJ3cyNVudUux378nN8Q9XBd; SameSite=None; Secure; path=/; expires=Sat, 06-Dec-25 10:03:39 GMT; HttpOnly
header: Set-Cookie: _cfuvid=5wqt6BmJgSTRr5DPv7RNglm94OQkTy0R3cOPHBVh9Xw-1765001019549-0.0.1.1-604800000; path=/; domain=.truthsocial.com; HttpOnly; Secure; SameSite=None
header: Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
header: X-Content-Type-Options: nosniff
header: Server: cloudflare
[debug] Extractor gave empty title. Creating a generic title
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec, channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 115670539534836133: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://static-assets-1.truthsocial.com/tmtg:prime-ts-assets/media_attachments/files/115/670/539/187/156/759/original/78d7d719fa88acd9.mp4"
[download] Truth video #115670539534836133 [115670539534836133].mp4 has already been downloaded
[download] 100% of    5.15MiB
```


Not working, Ubuntu VM:

```
[debug] Command-line config: ['https://truthsocial.com/@realDonaldTrump/posts/115670539534836133', '-v', '--print-traffic']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.11.12 from yt-dlp/yt-dlp [335653be8] (pip)
[debug] Python 3.13.10 (CPython x86_64 64bit) - Linux-6.8.0-88-generic-x86_64-with-glibc2.39 (OpenSSL 3.0.13 30 Jan 2024, glibc 2.39)
[debug] exe versions: ffmpeg N-121416-g6f1ab828d3-20251010 (setts), ffprobe N-121416-g6f1ab828d3-20251010
[debug] Optional libraries: Cryptodome-3.23.0, brotli-1.2.0, certifi-2025.11.12, curl_cffi-0.13.0, mutagen-1.47.0, requests-2.32.5, urllib3-2.5.0, websockets-15.0.1, yt_dlp_ejs-0.3.1
[debug] JS runtimes: deno-2.5.6
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Plugin directories: none
[debug] Loaded 1844 extractors
[Truth] Extracting URL: https://truthsocial.com/@realDonaldTrump/posts/115670539534836133
[Truth] 115670539534836133: Downloading JSON metadata
director: Handler preferences for this request: urllib=0, requests=100, websockets=0, curl_cffi=-100
director: Checking if "requests" supports this request.
director: Sending request via "requests"
requests: Starting new HTTPS connection (1): truthsocial.com:443
send: b'GET /api/v1/statuses/115670539534836133 HTTP/1.1\r\nHost: truthsocial.com\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Language: en-us,en;q=0.5\r\nSec-Fetch-Mode: navigate\r\nAccept-Encoding: gzip, deflate, br\r\nConnection: keep-alive\r\n\r\n'
reply: 'HTTP/1.1 403 Forbidden\r\n'
header: Date: Sat, 06 Dec 2025 06:04:22 GMT
header: Content-Type: text/html; charset=UTF-8
header: Transfer-Encoding: chunked
header: Connection: keep-alive
header: X-Frame-Options: SAMEORIGIN
header: Referrer-Policy: same-origin
header: Cache-Control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
header: Expires: Thu, 01 Jan 1970 00:00:01 GMT
header: Set-Cookie: __cf_bm=ufkfDMTy0S9F18HB6Rb3gafjizXQb8SNwb5o_9Z3AIQ-1765001062-1.0.1.1-3w12uC6jioPzNQ3qVSp2UGv9kAQIjm6ZcCImyQYlvrti4nap_4Tp_a8RqFY4XpdL8uRFrkPx1ggO45XUGul9Db1RZE3XGBjnTChHznEJML4; path=/; expires=Sat, 06-Dec-25 06:34:22 GMT; domain=.truthsocial.com; HttpOnly; Secure; SameSite=None
header: Vary: Accept-Encoding
header: Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
header: X-Content-Type-Options: nosniff
header: Server: cloudflare
header: CF-RAY: 9a9995e01bdb7f0a-YVR
header: Content-Encoding: gzip
ERROR: [Truth] 115670539534836133: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: Forbidden>)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 762, in extract
    ie_result = self._real_extract(url)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/truth.py", line 54, in _real_extract
    status = self._download_json(f'https://truthsocial.com/api/v1/statuses/{video_id}', video_id)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 1150, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 1108, in download_handle
    res = self._download_webpage_handle(
        url_or_request, video_id, note=note, errnote=errnote, fatal=fatal, encoding=encoding,
        data=data, headers=headers, query=query, expected_status=expected_status,
        impersonate=impersonate, require_impersonation=require_impersonation)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 973, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data,
                                 headers=headers, query=query, expected_status=expected_status,
                                 impersonate=impersonate, require_impersonation=require_impersonation)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 922, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 909, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query, extensions))
           ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/YoutubeDL.py", line 4252, in urlopen
    return self._request_director.send(req)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/common.py", line 117, in send
    response = handler.send(request)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/_helper.py", line 194, in wrapper
    return func(self, *args, **kwargs)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/common.py", line 359, in send
    return self._send(request)
           ~~~~~~~~~~^^^^^^^^^
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/_requests.py", line 361, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```

**Comment by gamer191** [comment]
@TheFiZi what happens if you do it on your Ubuntu VM with `--compat-options prefer-legacy-http-handler`

**Comment by Vangelis66** [comment]
This is just a hunch 😉 , but I think this is another case of the `rumble.com` extractor issue #15089; @TheFiZi 's **MacBook Air M4** has this:
```console
[debug] Python 3.13.10 (CPython arm64 64bit) - macOS-26.1-arm64-arm-64bit-Mach-O (OpenSSL 3.6.0 1 Oct 2025)
```
i.e. OpenSSL 3.**6.0**, thus it isn't being blocked by evil CF; whereas the **Ubuntu VM** has: 
```console
[debug] Python 3.13.10 (CPython x86_64 64bit) - Linux-6.8.0-88-generic-x86_64-with-glibc2.39 (OpenSSL 3.0.13 30 Jan 2024, glibc 2.39)
```
i.e. OpenSSL 3.**0.13**, which, as has been _detailed in the linked issue_, is one of the **CF-blocked versions** (via TLS-fingerprintig) ...

I have tested this theory myself on **Windows** and it's correct; logs that prove it: 

**Python 3.14.0 + OpenSSL-3.0.18**: 

<details>
<summary>yt-dlp_x86 -vF "https://truthsocial.com/@realDonaldTrump/posts/115670539534836133"</summary>

```console
[debug] Command-line config: ['-vF', 'https://truthsocial.com/@realDonaldTrump/posts/115670539534836133']
[debug] Encodings: locale cp1253, fs utf-8, pref cp1253, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version nicolaasjan/yt-dlp@2025.11.25.144622 (win7_x86_exe*)
[debug] Python 3.14.0 (CPython x86 32bit) - Windows-Vista-6.0.6003-SP2 (OpenSSL3.0.18 30 Sep 2025)
[debug] exe versions: ffmpeg 5.0 (fdk,setts)
[debug] Optional libraries: Cryptodome-3.23.0, brotli-1.2.0, certifi-2025.11.12, mutagen-1.47.0, requests-2.32.5, sqlite3-3.50.4, urllib3-2.5.0, websockets-15.0.1, yt_dlp_ejs-0.3.1
[debug] JS runtimes: none
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Plugin directories: none
[debug] Loaded 1853 extractors
[Truth] Extracting URL: https://truthsocial.com/@realDonaldTrump/posts/115670539534836133
[Truth] 115670539534836133: Downloading JSON metadata
ERROR: [Truth] 115670539534836133: Unable to download JSON metadata: HTTP Error403: Forbidden (caused by <HTTPError 403: Forbidden>)
  File "yt_dlp\extractor\common.py", line 762, in extract
  File "yt_dlp\extractor\truth.py", line 54, in _real_extract
  File "yt_dlp\extractor\common.py", line 1150, in download_content
  File "yt_dlp\extractor\common.py", line 1108, in download_handle
  File "yt_dlp\extractor\common.py", line 973, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 922, in _request_webpage

  File "yt_dlp\extractor\common.py", line 909, in _request_webpage
  File "yt_dlp\YoutubeDL.py", line 4252, in urlopen
  File "yt_dlp\networking\common.py", line 117, in send
  File "yt_dlp\networking\_helper.py", line 194, in wrapper
  File "yt_dlp\networking\common.py", line 359, in send
  File "yt_dlp\networking\_requests.py", line 361, in _send
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```
</details>

**Python 3.11.14 + OpenSSL-3.5.4**

<details>
<summary>yt-dlp -vF "https://truthsocial.com/@realDonaldTrump/posts/115670539534836133"</summary>

```console
[debug] Command-line config: ['--ffmpeg-location', 'FFmpeg', '--downloader-args', 'ffmpeg:-v 8 -stats', '-vF', 'https://truthsocial.com/@realDonaldTrump/posts/115670539534836133']
[debug] Encodings: locale cp1253, fs utf-8, pref cp1253, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version master@2025.12.05.231816 from yt-dlp/yt-dlp-master-builds [36b29bb35] (zip)
[debug] Python 3.11.14 (CPython x86 32bit) - Windows-Vista-6.0.6003-SP2 (OpenSSL 3.5.4 30 Sep 2025)
[debug] exe versions: ffmpeg n8.1-dev-630-N-121254-g635cb45 (setts), ffprobe n8.1-dev-630-N-121254-g635cb45
[debug] Optional libraries: Cryptodome-3.23.0, brotli-1.2.0, certifi-2025.11.12, curl_cffi-0.11.1, mutagen-1.47.0, requests-2.32.5, sqlite3-3.51.1, urllib3-2.5.0, websockets-15.0.1, yt_dlp_ejs-0.3.1
[debug] JS runtimes: none
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Plugin directories: none
[debug] Loaded 1853 extractors
[Truth] Extracting URL: https://truthsocial.com/@realDonaldTrump/posts/115670539534836133
[Truth] 115670539534836133: Downloading JSON metadata
[debug] Extractor gave empty title. Creating a generic title
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec, channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 115670539534836133:
ID EXT RESOLUTION | PROTO | VCODEC  ACODEC
-------------------------------------------
0  mp4 unknown    | https | unknown unknown
```
</details>

@TheFiZi : ~~Either~~ **update your OpenSSL version** ~~or, if that is not practical/possible, use the `--legacy-server-connect` switch~~ 😜 ...

**Later edit**: ` --legacy-server-connect` doesn't seem to work with TS 😿 (nor does ` --compat-options prefer-legacy-http-handler`), when your OpenSSL version is being blocked...

**Comment by bashonly** [comment]
Doesn't `--impersonate chrome` work?

**Comment by Vangelis66** [comment]
> Doesn't `--impersonate chrome` work?

... If the question was directed _at me_, am afraid I can't answer it myself, as being "OS-challenged" 😜 (Windows Vista SP2 **32-bit**); the "optional" `yt-dlp` dependency `curl_cffi`, in its latest (supported) version, `0.13.0`, 

a) only provides **64-bit** **Windows** wheels

https://pypi.org/project/curl-cffi/#files

b) more **importantly**, the lib it is a wrapper for, `curl-impersonate`, requires **at minimum Win7** to function properly (due to it being compiled with Google's **BoringSSL** as its crypto lib); any further discussion by me would be OT here, as `yt-dlp` _officially_ supports Win**8+**...

**TL;DR**: Someone else would have to check that...

**Comment by TheFiZi** [comment]
> [@TheFiZi](https://github.com/TheFiZi) what happens if you do it on your Ubuntu VM with `--compat-options prefer-legacy-http-handler`

<details>
<summary>Does not work</summary>

```
[debug] Command-line config: ['--compat-options', 'prefer-legacy-http-handler', '-v', '--print-traffic', 'https://truthsocial.com/@realDonaldTrump/posts/115670539534836133']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.11.12 from yt-dlp/yt-dlp [335653be8] (pip)
[debug] Compatibility options: prefer-legacy-http-handler
[debug] Python 3.13.10 (CPython x86_64 64bit) - Linux-6.8.0-88-generic-x86_64-with-glibc2.39 (OpenSSL 3.0.13 30 Jan 2024, glibc 2.39)
[debug] exe versions: ffmpeg N-121416-g6f1ab828d3-20251010 (setts), ffprobe N-121416-g6f1ab828d3-20251010
[debug] Optional libraries: Cryptodome-3.23.0, brotli-1.2.0, certifi-2025.11.12, curl_cffi-0.13.0, mutagen-1.47.0, requests-2.32.5, urllib3-2.5.0, websockets-15.0.1, yt_dlp_ejs-0.3.1
[debug] JS runtimes: deno-2.5.6
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Plugin directories: none
[debug] Loaded 1844 extractors
[Truth] Extracting URL: https://truthsocial.com/@realDonaldTrump/posts/115670539534836133
[Truth] 115670539534836133: Downloading JSON metadata
director: Handler preferences for this request: urllib=500, requests=100, websockets=0, curl_cffi=-100
director: Checking if "urllib" supports this request.
director: Sending request via "urllib"
send: b'GET /api/v1/statuses/115670539534836133 HTTP/1.1\r\nHost: truthsocial.com\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/137.0.0.0 Safari/537.36\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Language: en-us,en;q=0.5\r\nSec-Fetch-Mode: navigate\r\nAccept-Encoding: gzip, deflate, br\r\nConnection: close\r\n\r\n'
reply: 'HTTP/1.1 403 Forbidden\r\n'
header: Date: Sat, 06 Dec 2025 21:33:34 GMT
header: Content-Type: text/html; charset=UTF-8
header: Transfer-Encoding: chunked
header: Connection: close
header: X-Frame-Options: SAMEORIGIN
header: Referrer-Policy: same-origin
header: Cache-Control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
header: Expires: Thu, 01 Jan 1970 00:00:01 GMT
header: Set-Cookie: __cf_bm=vl1w7qxgh2noPjpF0a.mq1XdHRCj2fmfdTb9hdEDc54-1765056814-1.0.1.1-cCPaBG6zrpfEfpHH_aKXP7jBHACWzkpl9L5mN.ttUXvlrDBbKnzWpanqHVeF_10fkP1ZTQWAwBtJtoFRGZQ_OKBtGXgXjZnCi49YDh.dF_g; path=/; expires=Sat, 06-Dec-25 22:03:34 GMT; domain=.truthsocial.com; HttpOnly; Secure; SameSite=None
header: Vary: Accept-Encoding
header: Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
header: X-Content-Type-Options: nosniff
header: Server: cloudflare
header: CF-RAY: 9a9ee701ee532afc-YVR
header: Content-Encoding: gzip
ERROR: [Truth] 115670539534836133: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: Forbidden>)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 762, in extract
    ie_result = self._real_extract(url)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/truth.py", line 54, in _real_extract
    status = self._download_json(f'https://truthsocial.com/api/v1/statuses/{video_id}', video_id)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 1150, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 1108, in download_handle
    res = self._download_webpage_handle(
        url_or_request, video_id, note=note, errnote=errnote, fatal=fatal, encoding=encoding,
        data=data, headers=headers, query=query, expected_status=expected_status,
        impersonate=impersonate, require_impersonation=require_impersonation)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 973, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data,
                                 headers=headers, query=query, expected_status=expected_status,
                                 impersonate=impersonate, require_impersonation=require_impersonation)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 922, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/_urllib.py", line 417, in _send
    res = opener.open(urllib_req, timeout=self._calculate_timeout(request))
  File "/home/me/apps/python/lib/python3.13/urllib/request.py", line 495, in open
    response = meth(req, response)
  File "/home/me/apps/python/lib/python3.13/urllib/request.py", line 604, in http_response
    response = self.parent.error(
        'http', request, response, code, msg, hdrs)
  File "/home/me/apps/python/lib/python3.13/urllib/request.py", line 533, in error
    return self._call_chain(*args)
           ~~~~~~~~~~~~~~~~^^^^^^^
  File "/home/me/apps/python/lib/python3.13/urllib/request.py", line 466, in _call_chain
    result = func(*args)
  File "/home/me/apps/python/lib/python3.13/urllib/request.py", line 613, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/extractor/common.py", line 909, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query, extensions))
           ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/YoutubeDL.py", line 4252, in urlopen
    return self._request_director.send(req)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/common.py", line 117, in send
    response = handler.send(request)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/_helper.py", line 194, in wrapper
    return func(self, *args, **kwargs)
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/common.py", line 359, in send
    return self._send(request)
           ~~~~~~~~~~^^^^^^^^^
  File "/home/me/venv/thegoose/lib/python3.13/site-packages/yt_dlp/networking/_urllib.py", line 422, in _send
    raise HTTPError(UrllibResponseAdapter(e.fp), redirect_loop='redirect error' in str(e)) from e
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```
</details>

> Doesn't `--impersonate chrome` work?


<details>
<summary>Does work, but only via CLI</summary>

```
[debug] Command-line config: ['--compat-options', 'prefer-legacy-http-handler', '--impersonate', 'chrome', '-v', '--print-traffic', 'https://truthsocial.com/@realDonaldTrump/posts/115670539534836133']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.11.12 from yt-dlp/yt-dlp [335653be8] (pip)
[debug] Compatibility options: prefer-legacy-http-handler
[debug] Python 3.13.10 (CPython x86_64 64bit) - Linux-6.8.0-88-generic-x86_64-with-glibc2.39 (OpenSSL 3.0.13 30 Jan 2024, glibc 2.39)
[debug] exe versions: ffmpeg N-121416-g6f1ab828d3-20251010 (setts), ffprobe N-121416-g6f1ab828d3-20251010
[debug] Optional libraries: Cryptodome-3.23.0, brotli-1.2.0, certifi-2025.11.12, curl_cffi-0.13.0, mutagen-1.47.0, requests-2.32.5, urllib3-2.5.0, websockets-15.0.1, yt_dlp_ejs-0.3.1
[debug] JS runtimes: deno-2.5.6
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Plugin directories: none
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
[debug] Loaded 1844 extractors
[Truth] Extracting URL: https://truthsocial.com/@realDonaldTrump/posts/115670539534836133
[Truth] 115670539534836133: Downloading JSON metadata
director: Handler preferences for this request: urllib=500, requests=100, websockets=0, curl_cffi=900
director: Checking if "curl_cffi" supports this request.
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
director: Sending request via "curl_cffi"
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
* Host truthsocial.com:443 was resolved.
* IPv6: (none)
* IPv4: 172.64.148.163, 104.18.39.93
*   Trying 172.64.148.163:443...
* Cipher selection: TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA:AES256-SHA
* ALPS: offers h2
* ECH: will GREASE ClientHello
* ALPN: curl offers h2,http/1.1
*  CAfile: /home/sa_schewee/venv/thegoose/lib/python3.13/site-packages/certifi/cacert.pem
*  CApath: /etc/ssl/certs
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=truthsocial.com
*  start date: Nov  2 01:25:42 2025 GMT
*  expire date: Jan 31 02:25:39 2026 GMT
*  subjectAltName: host "truthsocial.com" matched cert's "truthsocial.com"
*  issuer: C=US; O=Google Trust Services; CN=WE1
*  SSL certificate verify ok.
* Connected to truthsocial.com (172.64.148.163) port 443
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://truthsocial.com/api/v1/statuses/115670539534836133
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:authority: truthsocial.com]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:path: /api/v1/statuses/115670539534836133]
* [HTTP/2] [1] [sec-ch-ua: "Chromium";v="136", "Google Chrome";v="136", "Not.A/Brand";v="99"]
* [HTTP/2] [1] [sec-ch-ua-mobile: ?0]
* [HTTP/2] [1] [sec-ch-ua-platform: "macOS"]
* [HTTP/2] [1] [upgrade-insecure-requests: 1]
* [HTTP/2] [1] [user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36]
* [HTTP/2] [1] [accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7]
* [HTTP/2] [1] [sec-fetch-site: none]
* [HTTP/2] [1] [sec-fetch-mode: navigate]
* [HTTP/2] [1] [sec-fetch-user: ?1]
* [HTTP/2] [1] [sec-fetch-dest: document]
* [HTTP/2] [1] [accept-encoding: gzip, deflate, br, zstd]
* [HTTP/2] [1] [accept-language: en-US,en;q=0.9]
* [HTTP/2] [1] [priority: u=0, i]
> GET /api/v1/statuses/115670539534836133 HTTP/2
Host: truthsocial.com
sec-ch-ua: "Chromium";v="136", "Google Chrome";v="136", "Not.A/Brand";v="99"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "macOS"
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Sec-Fetch-Site: none
Sec-Fetch-Mode: navigate
Sec-Fetch-User: ?1
Sec-Fetch-Dest: document
Accept-Encoding: gzip, deflate, br, zstd
Accept-Language: en-US,en;q=0.9
Priority: u=0, i

* Request completely sent off
< HTTP/2 200
< date: Sat, 06 Dec 2025 21:34:21 GMT
< content-type: application/json
< content-length: 1127
< cf-ray: 9a9ee8264a2f5a49-YVR
< content-encoding: gzip
< vary: Accept-Encoding
< content-security-policy: base-uri 'none'; default-src 'none'; font-src 'self'; img-src 'self' https: data: blob: https://static-assets-1.truthsocial.com; media-src 'self' https: data: blob: https://static-assets-1.truthsocial.com; style-src 'self' 'unsafe-inline'; frame-src 'self' https:; manifest-src 'self'; connect-src 'self' data: blob: wss://truthsocial.com https://truthsocial.com wss://links.truthsocial.com https://links.truthsocial.com wss://uploads.truthsocial.com https://uploads.truthsocial.com https://*.rmbl.ws https://*.ingest.sentry.io https://*.luckyorange.com https://pubsub.googleapis.com wss://*.visitors.live wss://*.luckyorange.com *.mixpanel.com *.mxpnl.com https://*.revcontent.com https://*.criteo.com https://*.criteo.net https://challenges.cloudflare.com https://*.perception.tv https://vstream.truthsocial.com https://*.truthsocial.tv https://*.googlesyndication.com https://*.adtrafficquality.google https://payments.braintree-api.com/graphql https://api.braintreegateway.com/* https://consent.cookie-script.com; script-src 'self' https://tools.luckyorange.com https://*.mxpnl.com https://assets.revcontent.com https://polyfill.io https://ads.pubmatic.com https://*.criteo.com https://challenges.cloudflare.com https://*.googlesyndication.com https://connect.facebook.net https://*.us.tvsquared.com https://googletagmanager.com https://*.adtrafficquality.google https://cdn.cookie-script.com; child-src 'self'; worker-src 'self' blob:; frame-ancestors 'none'
< cf-cache-status: DYNAMIC
* Added cookie __cf_bm="3XSX9ASR99Bedtx4DXy6heWv295VUmSVxGr6xV0HWmQ-1765056861-1.0.1.1-fe9UpZJ8mWyGxYT3LvdJxAgjctoOi99QIsxWuSN4kAsB65RZN3PNuSbNIYomNgQSHF0zvWhaTf1C45x8Rnz65E9U2e1RBg4e71Gw2OIz9go" for domain truthsocial.com, path /, expire 1765058661
< set-cookie: __cf_bm=3XSX9ASR99Bedtx4DXy6heWv295VUmSVxGr6xV0HWmQ-1765056861-1.0.1.1-fe9UpZJ8mWyGxYT3LvdJxAgjctoOi99QIsxWuSN4kAsB65RZN3PNuSbNIYomNgQSHF0zvWhaTf1C45x8Rnz65E9U2e1RBg4e71Gw2OIz9go; path=/; expires=Sat, 06-Dec-25 22:04:21 GMT; domain=.truthsocial.com; HttpOnly; Secure; SameSite=None
* Added cookie __cflb="0H28vTPqhjwKvpvovPffsJ3cyNVudUuwmLNk1W98beF" for domain truthsocial.com, path /, expire 1765071261
< set-cookie: __cflb=0H28vTPqhjwKvpvovPffsJ3cyNVudUuwmLNk1W98beF; SameSite=None; Secure; path=/; expires=Sun, 07-Dec-25 01:34:21 GMT; HttpOnly
* Added cookie _cfuvid="zx4W5UpMAYnDsnd7_IlcxsfnsoPI2fDDCJ5QhMiMHrs-1765056861250-0.0.1.1-604800000" for domain truthsocial.com, path /, expire 0
< set-cookie: _cfuvid=zx4W5UpMAYnDsnd7_IlcxsfnsoPI2fDDCJ5QhMiMHrs-1765056861250-0.0.1.1-604800000; path=/; domain=.truthsocial.com; HttpOnly; Secure; SameSite=None
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< x-content-type-options: nosniff
< server: cloudflare
<
* Connection #0 to host truthsocial.com left intact
[debug] Extractor gave empty title. Creating a generic title
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec, channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 115670539534836133: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://static-assets-1.truthsocial.com/tmtg:prime-ts-assets/media_attachments/files/115/670/539/187/156/759/original/78d7d719fa88acd9.mp4"
director: Handler preferences for this request: urllib=500, requests=100, websockets=0, curl_cffi=900
director: Checking if "curl_cffi" supports this request.
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
director: Sending request via "curl_cffi"
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
curl_cffi: resolved impersonate target chrome to chrome-136:macos-15
* Added cookie __cf_bm="3XSX9ASR99Bedtx4DXy6heWv295VUmSVxGr6xV0HWmQ-1765056861-1.0.1.1-fe9UpZJ8mWyGxYT3LvdJxAgjctoOi99QIsxWuSN4kAsB65RZN3PNuSbNIYomNgQSHF0zvWhaTf1C45x8Rnz65E9U2e1RBg4e71Gw2OIz9go" for domain truthsocial.com, path /, expire 1765058661
* Added cookie _cfuvid="zx4W5UpMAYnDsnd7_IlcxsfnsoPI2fDDCJ5QhMiMHrs-1765056861250-0.0.1.1-604800000" for domain truthsocial.com, path /, expire 0
* Added cookie __cflb="0H28vTPqhjwKvpvovPffsJ3cyNVudUuwmLNk1W98beF" for domain truthsocial.com, path /, expire 1765071261
* Host static-assets-1.truthsocial.com:443 was resolved.
* IPv6: (none)
* IPv4: 104.18.39.93, 172.64.148.163
*   Trying 104.18.39.93:443...
* Cipher selection: TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA:AES256-SHA
* ALPS: offers h2
* ECH: will GREASE ClientHello
* ALPN: curl offers h2,http/1.1
*  CAfile: /home/sa_schewee/venv/thegoose/lib/python3.13/site-packages/certifi/cacert.pem
*  CApath: /etc/ssl/certs
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=truthsocial.com
*  start date: Nov  2 01:25:42 2025 GMT
*  expire date: Jan 31 02:25:39 2026 GMT
*  subjectAltName: host "static-assets-1.truthsocial.com" matched cert's "*.truthsocial.com"
*  issuer: C=US; O=Google Trust Services; CN=WE1
*  SSL certificate verify ok.
* Connected to static-assets-1.truthsocial.com (104.18.39.93) port 443
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://static-assets-1.truthsocial.com/tmtg:prime-ts-assets/media_attachments/files/115/670/539/187/156/759/original/78d7d719fa88acd9.mp4
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:authority: static-assets-1.truthsocial.com]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:path: /tmtg:prime-ts-assets/media_attachments/files/115/670/539/187/156/759/original/78d7d719fa88acd9.mp4]
* [HTTP/2] [1] [cookie: _cfuvid=zx4W5UpMAYnDsnd7_IlcxsfnsoPI2fDDCJ5QhMiMHrs-1765056861250-0.0.1.1-604800000; __cf_bm=3XSX9ASR99Bedtx4DXy6heWv295VUmSVxGr6xV0HWmQ-1765056861-1.0.1.1-fe9UpZJ8mWyGxYT3LvdJxAgjctoOi99QIsxWuSN4kAsB65RZN3PNuSbNIYomNgQSHF0zvWhaTf1C45x8Rnz65E9U2e1RBg4e71Gw2OIz9go]
* [HTTP/2] [1] [sec-ch-ua: "Chromium";v="136", "Google Chrome";v="136", "Not.A/Brand";v="99"]
* [HTTP/2] [1] [sec-ch-ua-mobile: ?0]
* [HTTP/2] [1] [sec-ch-ua-platform: "macOS"]
* [HTTP/2] [1] [upgrade-insecure-requests: 1]
* [HTTP/2] [1] [user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36]
* [HTTP/2] [1] [accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7]
* [HTTP/2] [1] [sec-fetch-site: none]
* [HTTP/2] [1] [sec-fetch-mode: navigate]
* [HTTP/2] [1] [sec-fetch-user: ?1]
* [HTTP/2] [1] [sec-fetch-dest: document]
* [HTTP/2] [1] [accept-encoding: identity]
* [HTTP/2] [1] [accept-language: en-US,en;q=0.9]
* [HTTP/2] [1] [priority: u=0, i]
> GET /tmtg:prime-ts-assets/media_attachments/files/115/670/539/187/156/759/original/78d7d719fa88acd9.mp4 HTTP/2
Host: static-assets-1.truthsocial.com
Cookie: _cfuvid=zx4W5UpMAYnDsnd7_IlcxsfnsoPI2fDDCJ5QhMiMHrs-1765056861250-0.0.1.1-604800000; __cf_bm=3XSX9ASR99Bedtx4DXy6heWv295VUmSVxGr6xV0HWmQ-1765056861-1.0.1.1-fe9UpZJ8mWyGxYT3LvdJxAgjctoOi99QIsxWuSN4kAsB65RZN3PNuSbNIYomNgQSHF0zvWhaTf1C45x8Rnz65E9U2e1RBg4e71Gw2OIz9go
sec-ch-ua: "Chromium";v="136", "Google Chrome";v="136", "Not.A/Brand";v="99"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "macOS"
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Sec-Fetch-Site: none
Sec-Fetch-Mode: navigate
Sec-Fetch-User: ?1
Sec-Fetch-Dest: document
Accept-Encoding: identity
Accept-Language: en-US,en;q=0.9
Priority: u=0, i

* Request completely sent off
< HTTP/2 200
< date: Sat, 06 Dec 2025 21:34:21 GMT
< content-type: video/quicktime
< content-length: 5399044
< cf-ray: 9a9ee8271d058905-YVR
< last-modified: Sat, 06 Dec 2025 03:39:22 GMT
< x-rgw-object-type: Normal
< x-amz-replication-status: PENDING
< etag: "01d270cdd1952fada746cfc1ba414f3f"
< cache-control: public, max-age=315576000, immutable
< x-amz-request-id: tx00000a0428c19e0f03d71-006933a56b-57296226-us-east-2
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< cf-cache-status: HIT
< age: 1478
< accept-ranges: bytes
< vary: Accept-Encoding
< x-content-type-options: nosniff
< server: cloudflare
<
[download] Destination: Truth video #115670539534836133 [115670539534836133].mp4
[download]  77.7% of    5.15MiB at   25.08MiB/s ETA 00:00* Connection #0 to host static-assets-1.truthsocial.com left intact
[download] 100% of    5.15MiB in 00:00:00 at 23.15MiB/s
```
</details>

I am trying to do this via the Python module, which now that I look at https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/YoutubeDL.py#L128-L278, it does not appear I can pass `--impersonate` or `--legacy-server-connect` as options like:

```
ydl_opts = {
    'impersonate': 'chrome',
    'compat-options': ['prefer-legacy-http-handler'],
}
```

I can very likely upgrade OpenSSL and try again, will just take me a bit to compile things heh.

**Comment by TheFiZi** [comment]
Upgrading OpenSSL to 3.5.x worked, all I need is:

```
ydl_opts = {
    'outtmpl': os.path.join(CDN_BASE_DIR, f'{uuid.uuid4()}.%(ext)s'),
    'format': 'mp4',
    'quiet': True,
}
```

**Comment by bashonly** [comment]
> I am trying to do this via the Python module

Don't use the compat option!

And *do* use [`devscripts/cli_to_api.py`](https://github.com/yt-dlp/yt-dlp/blob/master/devscripts/cli_to_api.py).

```py
from yt_dlp.networking.impersonate import ImpersonateTarget

ydl_opts = {
    'impersonate': ImpersonateTarget(client='chrome', version=None, os=None, os_version=None),
}
```

You were looking at the youtube-dl source code

**Comment by TheFiZi** [comment]
> You were looking at the youtube-dl source code

Ah shit, my bad.

Thank you for the link to that script. I suspect that is going to save me a ton of headache down the road.
